src/jvm/main/org/jetbrains/kotlinx/lincheck/Reporter.kt

```diff
@@ -597,22 +597,20 @@ internal data class ExceptionStackTracesResult(val exceptionStackTraces: Map<Thr
  */
 internal fun collectExceptionStackTraces(executionResult: ExecutionResult): ExceptionsProcessingResult {
     val exceptionStackTraces = mutableMapOf<Throwable, ExceptionNumberAndStacktrace>()
-
-    (executionResult.initResults.asSequence()
-            + executionResult.parallelResults.asSequence().flatten()
-            + executionResult.postResults.asSequence())
+    executionResult.allResults
         .filterIsInstance<ExceptionResult>()
         .forEachIndexed { index, exceptionResult ->
             val exception = exceptionResult.throwable
-
-            val filteredStacktrace = exception.stackTrace.takeWhile { LINCHECK_PACKAGE_NAME !in it.className }
-            if (filteredStacktrace.isEmpty()) { // Exception in Lincheck itself
+            val isInternalLincheckBug = exception.stackTrace?.firstOrNull()
```

Comment:
Could you please explain what should happen here and add it as a comment?

I guess, we should filter out only the deepest occurrences of the Lincheck internals, but I might be wrong here.

Reply:
There are two orthogonal questions here:
1. how we decide if the exception is Lincheck-internal or from the user code
2. in case of user-code exception, how do we filter the stack trace. 

For (1), I suppose the most robust solution is to check the latest stack trace element, to see if the exception was raised from some Lincheck method. This is what implemented here. 

For (2), both options (filtering all Lincheck methods from stack trace, or filtering only the deepest one) seem valid. For me, the option to filter all Lincheck methods from stack trace looks more intuitive, but I do not insist on it.

Reply:
>  I suppose the most robust solution is to check the latest stack trace element
This approach might not work when the exception is thrown in an `ArrayList` implementation, for example.

The second approach filters out the test running mechanism but keeps the model checking strategy internals when an exception is thrown from there.

Reply:
> This approach might not work when the exception is thrown in an `ArrayList` implementation, for example.

But in this case, the previous code would not work as well (in a sense that it would not detect this as a Lincheck bug).

Reply:
Let's proceed with this strategy and keep the discussion unresolved so we can easily find it if needed.